In [1]:
#PROJECT & ORGANIZATION CHANGE MANAGEMENT TEST
#ABAHO BOAZ
#lets import libralies for our codes
import pandas as pd
import numpy as np

In [2]:
#loading data set
df=pd.read_csv("COVID_19 AMERICA 7.10.2021 1810PM.csv", sep=",")
df

Name WHO Region  Cases - cumulative total  \
0                              Bahamas   Americas                     12407   
1             Turks and Caicos Islands   Americas                      2424   
2                  Antigua and Barbuda   Americas                      1263   
3                     Saint Barthélemy   Americas                      1040   
4               British Virgin Islands   Americas                       295   
5                             Dominica   Americas                       192   
6             United States of America   Americas                  33257768   
7                               Brazil   Americas                  18169881   
8                            Argentina   Americas                   4326101   
9                             Colombia   Americas                   4027016   
10                              Mexico   Americas                   2487747   
11                                Peru   Americas                   2036449   
12                               Chile   Americas                   1531872   
13                              Canada   Americas                   1410927   
14                             Ecuador   Americas                    449483   
15    Bolivia (Plurinational State of)   Americas                    426748   
16                            Paraguay   Americas                    411615   
17                              Panama   Americas                    396526   
18                             Uruguay   Americas                    360247   
19                          Costa Rica   Americas                    357523   
20                  Dominican Republic   Americas                    320136   
21                           Guatemala   Americas                    284741   
22  Venezuela (Bolivarian Republic of)   Americas                    264551   
23                            Honduras   Americas                    255663   
24                                Cuba   Americas                    174789   
25                         Puerto Rico   Americas                    139802   
26                         El Salvador   Americas                     77484   
27                             Jamaica   Americas                     49841   
28                 Trinidad and Tobago   Americas                     31207   
29                       French Guiana   Americas                     27040   
30                            Suriname   Americas                     20549   
31                              Guyana   Americas                     19359   
32                               Haiti   Americas                     17963   
33                          Guadeloupe   Americas                     17427   
34                              Belize   Americas                     13083   
35                             Curaçao   Americas                     12322   
36                          Martinique   Americas                     12199   
37                               Aruba   Americas                     11118   
38                           Nicaragua   Americas                      6402   
39                         Saint Lucia   Americas                      5238   
40                            Barbados   Americas                      4052   
41        United States Virgin Islands   Americas                      3819   
42                        Sint Maarten   Americas                      2585   
43                        Saint Martin   Americas                      2314   
44    Saint Vincent and the Grenadines   Americas                      2207   
45                             Bonaire   Americas                      1608   
46               Saint Kitts and Nevis   Americas                       396   
47                             Grenada   Americas                       161   
48                            Anguilla   Americas                       109   
49                          Montserrat   Americas                        20   
50   

In [3]:
df.head()

Name WHO Region  Cases - cumulative total  \
0                   Bahamas   Americas                     12407   
1  Turks and Caicos Islands   Americas                      2424   
2       Antigua and Barbuda   Americas                      1263   
3          Saint Barthélemy   Americas                      1040   
4    British Virgin Islands   Americas                       295   

   Cases - cumulative total per 100000 population  \
0                                         3155.04   
1                                         6260.65   
2                                         1289.71   
3                                        10520.99   
4                                          975.63   

   Cases - newly reported in last 7 days  \
0                                    112   
1                                      1   
2                                      0   
3                                      8   
4                                      0   

   Cases - newly reported in last 7 days per 100000 population  \
0                                              28.48             
1                                               2.58             
2                                               0.00             
3                                              80.93             
4                                               0.00             

   Cases - newly reported in last 24 hours  Deaths - cumulative total  \
0                                        0                        244   
1                                        0                         18   
2                                        0                         42   
3                                        0                          1   
4                                        0                          1   

   Deaths - cumulative total per 100000 population  \
0                                            62.05   
1                                            46.49   
2                                            42.89   
3                                            10.12   
4                                             3.31   

   Deaths - newly reported in last 7 days  \
0                                       3   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   Deaths - newly reported in last 7 days per 100000 population  \
0                                               0.76              
1                                               0.00              
2                                               0.00              
3                                               0.00              
4                                               0.00              

   Deaths - newly reported in last 24 hours Transmission Classification  
0                                         0           Clusters of cases  
1                                         0           Clusters of cases  
2                                         0           Clusters of cases  
3                                         0           Clusters of cases  
4                                         0           Clusters of cases

In [4]:
df.tail()

Name WHO Region  Cases - cumulative total  \
51                         Saba   Americas                         7   
52                      Bermuda   Americas                      2504   
53               Cayman Islands   Americas                       609   
54  Falkland Islands (Malvinas)   Americas                        60   
55    Saint Pierre and Miquelon   Americas                        26   

    Cases - cumulative total per 100000 population  \
51                                          362.13   
52                                         4021.00   
53                                          926.66   
54                                         1722.65   
55                                          448.66   

    Cases - newly reported in last 7 days  \
51                                      0   
52                                      5   
53                                     10   
54                                      0   
55                                      0   

    Cases - newly reported in last 7 days per 100000 population  \
51                                               0.00             
52                                               8.03             
53                                              15.22             
54                                               0.00             
55                                               0.00             

    Cases - newly reported in last 24 hours  Deaths - cumulative total  \
51                                        0                          0   
52                                        0                         33   
53                                        0                          2   
54                                        0                          0   
55                                        0                          0   

    Deaths - cumulative total per 100000 population  \
51                                             0.00   
52                                            52.99   
53                                             3.04   
54                                             0.00   
55                                             0.00   

    Deaths - newly reported in last 7 days  \
51                                       0   
52                                       0   
53                                       0   
54                                       0   
55                                       0   

    Deaths - newly reported in last 7 days per 100000 population  \
51                                                0.0              
52                                                0.0              
53                                                0.0              
54                                                0.0              
55                                                0.0              

    Deaths - newly reported in last 24 hours Transmission Classification  
51                                         0                    No cases  
52                                         0              Sporadic cases  
53                                         0              Sporadic cases  
54                                         0              Sporadic cases  
55                                         0              Sporadic cases

In [5]:
df.dtypes

Name                                                             object
WHO Region                                                       object
Cases - cumulative total                                          int64
Cases - cumulative total per 100000 population                  float64
Cases - newly reported in last 7 days                             int64
Cases - newly reported in last 7 days per 100000 population     float64
Cases - newly reported in last 24 hours                           int64
Deaths - cumulative total                                         int64
Deaths - cumulative total per 100000 population                 float64
Deaths - newly reported in last 7 days                            int64
Deaths - newly reported in last 7 days per 100000 population    float64
Deaths - newly reported in last 24 hours                          int64
Transmission Classification                                      object
dtype: object

In [6]:
df.describe()

Cases - cumulative total  \
count              5.600000e+01   
mean               1.275874e+06   
std                5.038072e+06   
min                7.000000e+00   
25%                2.057250e+03   
50%                1.525500e+04   
75%                3.294828e+05   
max                3.325777e+07   

       Cases - cumulative total per 100000 population  \
count                                       56.000000   
mean                                      3935.451429   
std                                       3243.103255   
min                                         96.640000   
25%                                       1265.932500   
50%                                       3053.085000   
75%                                       6197.410000   
max                                      10520.990000   

       Cases - newly reported in last 7 days  \
count                              56.000000   
mean                            20456.053571   
std                             78014.394954   
min                                 0.000000   
25%                                 9.500000   
50%                               152.000000   
75%                              7444.250000   
max                            541293.000000   

       Cases - newly reported in last 7 days per 100000 population  \
count                                          56.000000             
mean                                           78.053571             
std                                           103.899707             
min                                             0.000000             
25%                                             8.027500             
50%                                            27.855000             
75%                                           101.517500             
max                                           395.310000             

       Cases - newly reported in last 24 hours  Deaths - cumulative total  \
count                                56.000000                  56.000000   
mean                               3853.892857               33542.107143   
std                               16162.765967              109808.602847   
min                                   0.000000                   0.000000   
25%                                   0.000000                  17.750000   
50%                                  15.000000                 255.000000   
75%                                1102.500000                5653.250000   
max                              115228.000000              597727.000000   

       Deaths - cumulative total per 100000 population  \
count                                        56.000000   
mean                                         74.064107   
std                                          92.853474   
min                                           0.000000   
25%                                          10.645000   
50%                                          51.180000   
75%                                          89.102500   
max                                         579.500000   

       Deaths - newly reported in last 7 days  \
count                               56.000000   
mean                               538.428571   
std                               1939.784410   
min                                  0.000000   
25%                                  0.000000   
50%                                  3.500000   
75%                                135.750000   
max                              13416.000000   

       Deaths - newly reported in last 7 days per 100000 population  \
count                                          56.000000              
mean                                            1.768929              
std                                             2.770508              
min                                             0.000000              
25%                                             0.000000              
50%                         

In [7]:
df.shape

(56, 13)

In [8]:
%%capture
!pip install seaborn

In [9]:
df['Transmission Classification'].value_counts().to_frame()

Transmission Classification
Community transmission                           41
Clusters of cases                                 6
No cases                                          5
Sporadic cases                                    4

# Simple Linear Regression

In [10]:
#import matplotlib.pyplot as plt
#import numpy as np
#%conda install sklearn
#%matplotlib inline
#from sklearn.linear_model import linearRegression

In [11]:
from sklearn.linear_model import LinearRegression

In [12]:
#Using identified cases for the next 24 hours
x=df[['Cases - newly reported in last 24 hours']]
y=df['Deaths - newly reported in last 24 hours']

In [13]:
lm=LinearRegression()
lm.fit(x,y)
yhat=lm.predict(x)
yhat[0:5]

array([9.95968105, 9.95968105, 9.95968105, 9.95968105, 9.95968105])

In [14]:
#calculation of coefficient m
lm.coef_

array([0.02106987])

In [15]:
#intercept c
lm.intercept_

9.95968105413516

In [16]:
#calculating r-squared
lm.score(x,y)

0.9855145590117684

y=mx+c
$$Deaths - newly reported in last 24 hours=0.02106987xCases - newly reported in last 24 hours+9.95968105413516$$


In [17]:
#Let's import the function mean_squared_error from the module metrics
from sklearn.metrics import mean_squared_error


In [18]:
#RMSE
y=lm.predict(x)
MSE = mean_squared_error(df['Deaths - newly reported in last 24 hours'], y)
print('MSE is', MSE)


MSE is 1674.166108155738


In [19]:
#calculate RSME
#import necessary libraries
from math import sqrt


In [20]:
sqrt(mean_squared_error(df['Deaths - newly reported in last 24 hours'], y))


40.9165749807549

# Multiple LinearRegression

Predicting Deaths-newly reported in last 24 hours in America by looking at; Cases - newly reported in last 7 days,cases - cumulative total per 100000 population, and Cases - newly reported in last 24 hours


Model formulae;
$$𝑦=𝑏1𝑥1+𝑏2𝑥2+b3x3+𝑐$$


In [21]:
# The different variables here are; Cases - newly reported in last 7 days, cases - cumulative total per 100000 population and Cases,Cases - newly reported in last 24 hours
z = df[['Cases - newly reported in last 7 days', 'Cases - cumulative total per 100000 population','Cases - newly reported in last 24 hours']]



In [22]:
#Fitting the linear model using the three above-mentioned variables.
lm.fit(z, df['Deaths - newly reported in last 24 hours'])


LinearRegression()

In [23]:
#calculation of coefficient b1,b2,b3 
lm.coef_

array([0.0002005 , 0.00085955, 0.02005241])

In [24]:
#intercept 
lm.intercept_

6.396806461731813

In [25]:
#calculating r-squared
lm.score(z, df['Deaths - newly reported in last 24 hours'])


0.9856121774537552

In [26]:
#Calculating the Mean Squared Error (MSE)
#prediction based on our model
y_predict_multifit = lm.predict(z)
#Now we can compare the predicted results with the actual results
mean_squared_error(df['Deaths - newly reported in last 24 hours'], y_predict_multifit)


1662.883780801116

In [27]:
#calculating RMSE
sqrt(mean_squared_error(df['Deaths - newly reported in last 24 hours'], y_predict_multifit))

40.778472026316976

# The best design model to predict the number of deaths due to Covid_19 pandemic in Americas Region is Multiple Linear Regression.

# Reason

# r-square for Multiple Linear Regression = 0.9856121774537552 which is Higher than the r-square for Simple LinearRegression = 0.9855145590117684